File: examples/interpreter-in-browser/www/ex_sample.ex
defmodule ExSample do
  # NOTE: this file is not automatically compiled to .erl, please check the
  # README.md file for more information.
  def run_me(arg) do
    :lumen_intrinsics.println("Doing the Lumen web work...")
    {:ok, window} = Lumen.Web.Window.window()
    {:ok, document} = Lumen.Web.Window.document(window)
    {:ok, paragraph} = Lumen.Web.Document.create_element(document, "p")

    text =
      Lumen.Web.Document.create_text_node(
        document,
        "This text was created through Elixir in your browser."
      )

    :ok = Lumen.Web.Node.append_child(paragraph, text)
    {:ok, output} = Lumen.Web.Document.get_element_by_id(document, "output")
    :ok = Lumen.Web.Node.append_child(output, paragraph)
    :lumen_intrinsics.println("Done!")
  end
end

File: examples/interpreter-in-browser/www/ex_sample.erl
-module('Elixir.ExSample').

-export(['__info__'/1, run_me/1]).

-spec '__info__'(attributes | compile | functions |
		 macros | md5 | module | deprecated) -> any().

'__info__'(module) -> 'Elixir.ExSample';
'__info__'(functions) -> [{run_me, 1}];
'__info__'(macros) -> [];
'__info__'(Key = attributes) ->
    erlang:get_module_info('Elixir.ExSample', Key);
'__info__'(Key = compile) ->
    erlang:get_module_info('Elixir.ExSample', Key);
'__info__'(Key = md5) ->
    erlang:get_module_info('Elixir.ExSample', Key);
'__info__'(deprecated) -> [].

run_me(_arg@1) ->
    lumen_intrinsics:println(<<"Doing the Lumen web work...">>),
    {ok, _window@1} = 'Elixir.Lumen.Web.Window':window(),
    {ok, _document@1} =
	'Elixir.Lumen.Web.Window':document(_window@1),
    {ok, _paragraph@1} =
	'Elixir.Lumen.Web.Document':create_element(_document@1,
						   <<"p">>),
    _text@1 =
	'Elixir.Lumen.Web.Document':create_text_node(_document@1,
						     <<"This text was created through Elixir "
						       "in your browser.">>),
    ok = 'Elixir.Lumen.Web.Node':append_child(_paragraph@1,
					      _text@1),
    {ok, _output@1} =
	'Elixir.Lumen.Web.Document':get_element_by_id(_document@1,
						      <<"output">>),
    ok = 'Elixir.Lumen.Web.Node':append_child(_output@1,
					      _paragraph@1),
    lumen_intrinsics:println(<<"Done!">>).
